src/main/java/org/recommendation/data/GenreDataInitializer.java

```diff
@@ -0,0 +1,36 @@
+package org.recommendation.data;
+
+import org.apache.poi.xssf.usermodel.XSSFWorkbook;
+import org.recommendation.log.Logger;
+import org.recommendation.log.SoutLogger;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+public class GenreDataInitializer extends DataInitializer {
+    public static HashMap<String, String> GenereMap = new HashMap<>();
+    private Logger logger = new SoutLogger();
+
+    @Override
+    public Map<String, String[]> readAndCleanData(final BufferedReader br) {
+        String line;
+        try {
+            while ((line = br.readLine()) != null) {
+                String[] data = line.split("\\|");
+                if (data.length > 1) {
+                    GenereMap.put(data[1], data[0]);
+                }
+            }
+        } catch (Exception exception) {
+            logger.error("GenreDataInitializer readAndCleanData()", exception);
+        }
+        return null;
```

Comment:
it will return null in each and every cases why?

Reply:
there is no need to implement writeData in case of genre, GenereMap is used inmemory wherever required so simply returning null in this case.